### PR TITLE
2216  setting data layer updates

### PR DESCRIPTION
**Pull request by JettTech**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -20,8 +20,10 @@
 .env.production.local
 tmp-storage
 tmp
-conductor-config.toml
 holofuel-conductor-config.toml
+example-conductor-config.toml
```
Comment (robbiecarlton) [inline .gitignore]: Don't think we should be ignoring the `example-conductor-config`. We want that in the repo as an example.

File: scripts/HHAProviderFlow/index.js [hunk 2]
```diff
@@ -1,36 +1,33 @@
+const fs = require('fs')
```
Comment (robbiecarlton) [inline scripts/HHAProviderFlow/index.js]: glad you updated this.

File: src/data-interfaces/HposInterface.js [hunk 3]
```diff
@@ -10,11 +10,13 @@ const axiosConfig = {
...
 
-export function hposCall (method = 'get', path) {
+export function hposCall (method = 'get', path, apiVersion = 'v1') {
   if (MOCK_HPOS_CONNECTION) {
-    return mockCallHpos(method, path)
+    return mockCallHpos(method, apiVersion, path)
   } else {
-    const fullPath = process.env.REACT_APP_HPOS_URL + '/' + path
+    const fullPath = process.env.REACT_APP_HPOS_URL + '/' + apiVersion + '/' + path
+    console.log('fullPath : ', fullPath)
```
Comment (robbiecarlton) [inline src/data-interfaces/HposInterface.js]: remove log

File: src/data-interfaces/HposInterface.js [hunk 4]
```diff
@@ -10,11 +10,13 @@ const axiosConfig = {
   }
 }
 
-export function hposCall (method = 'get', path) {
+export function hposCall (method = 'get', path, apiVersion = 'v1') {
```
Comment (robbiecarlton) [inline src/data-interfaces/HposInterface.js:13]: nice addition

**Review by robbiecarlton — CHANGES_REQUESTED** [review]
Added a couple of comments here but didn't complete the review because (afaict) there are multiple other branches in here and it got too confusing. I'll rereview once those other branches are merged.

File: .gitignore (hunk 1 above)
Comment (JettTech) [reply .gitignore]: We had removed this file, but I added it back when comparing the manual conductor with the nix auto-generated one, so I just added the file to the ignore list.  At any rate, I still agree that it's unnecessary overall, so I'll go ahead and remove this as we shouldn't really be needing it beyond the local comparisons I was testing out.

File: src/graphql-server/resolvers.js [hunk 5]
```diff
@@ -55,11 +55,13 @@ export const resolvers = {
...
 
-    hposSettings: HposInterface.os.settings
+    hposSettings: HposInterface.os.settings,
+
+    hposStatus: HposInterface.os.status
   },
 
   Mutation: {
-    registerUser: (_, userData) => dataMappedCall('person', userData, HyloDnaInterface.currentUser.create),
+    registerUser: (_, userData) => { dataMappedCall('person', userData, HyloDnaInterface.currentUser.create) },
```
Comment (robbiecarlton) [inline src/graphql-server/resolvers.js]: remove curly braces

File: src/graphql-server/schema.graphql [hunk 6]
```diff
@@ -76,7 +76,31 @@ type Ledger {
...
+  deviceName: String
+}
+
+type hydraStruct {
+  rev: String
+}
+
+type HposVersion {
+  availableVersion: hydraStruct
+  currentVersion: hydraStruct
```
Comment (robbiecarlton) [inline src/graphql-server/schema.graphql]: these can just be `Strings`

File: src/graphql-server/schema.graphql [hunk 7]
```diff
@@ -76,7 +76,31 @@ type Ledger {
...
+  hostName: String
+  registrationEmail: String
+  networkStatus: String # ie: 'live'
+  sshAccess: Boolean
+  deviceName: String
+}
+
+type hydraStruct {
+  rev: String
+}
```
Comment (robbiecarlton) [inline src/graphql-server/schema.graphql]: remove

File: src/graphql-server/schema.graphql [hunk 8]
```diff
@@ -103,10 +128,11 @@ type Mutation {
...
-  holofuelRequest(counterpartyId:String, amount:Float): Transaction
-  holofuelOffer(counterpartyId:String, amount:Float, requestId:ID): Transaction
+  holofuelRequest(counterpartyId: String, amount: Float): Transaction
+  holofuelOffer(counterpartyId: String, amount: Float, requestId:ID): Transaction
   holofuelAcceptOffer(transactionId: ID): Transaction 
   holofuelDecline(transactionId: ID): Transaction
   holofuelCancel(transactionId: ID): Transaction
-  hposUpdateSettings(version: String): HposSettings
+  hposUpdateSettings(hostPubKey: String, hostName: String, sshAccess: Boolean): HposSettings
+  hposUpdateVersion(availableVersion: String): HposStatus
```
Comment (robbiecarlton) [inline src/graphql-server/schema.graphql]: don't pass a param to this

File: src/graphql/HposStatusQuery.gql [hunk 9]
```diff
@@ -0,0 +1,7 @@
+query HposStatus {
+  hposStatus {
+    versionInfo { availableVersion { rev }, currentVersion { rev } }
+    networkId
+    ports { primaryPort }
```
Comment (robbiecarlton) [inline src/graphql/HposStatusQuery.gql]: use new lines for nested graphql queries

File: src/graphql/HposUpdateVersionMutation.gql [hunk 10]
```diff
@@ -0,0 +1,7 @@
+mutation HposUpdateVersion ($availableVersion: String) {
+  hposUpdateVersion (availableVersion: $availableVersion) {
+    versionInfo { availableVersion { rev }, currentVersion { rev } }
```
Comment (robbiecarlton) [inline src/graphql/HposUpdateVersionMutation.gql]: newlines

File: src/mock-dnas/hpos.js [hunk 11]
```diff
@@ -1,15 +1,74 @@
...
       return {
-        ...data.get['os-update'],
+        ...data.get['/v1/config'],
         ...params
       }
+    },
+
+    // Forces HoloPortOS upgrade. - Returns Status Code ONLY (200 if OK/Successful, 400 if Bad Request, or 401 if Unauthorized).
+    // TODO: Verify that NO PARAMS are needed.  Talk to HCen to see if they can return new Status Obj upon successful update.
+    '/v1/upgrade': params => {
```
Comment (robbiecarlton) [inline src/mock-dnas/hpos.js]: no params

File: src/mock-dnas/hpos.js [hunk 12]
```diff
@@ -1,15 +1,74 @@
...
   },
   post: {
-    'os-update': params => {
+    '/v1/config': params => {
+      data.get['/v1/config'] = { ...data.get['/v1/config'], ...params }
       return {
-        ...data.get['os-update'],
+        ...data.get['/v1/config'],
         ...params
       }
```
Comment (robbiecarlton) [inline src/mock-dnas/hpos.js]: `return data.get['/v1/config']`

**Review by robbiecarlton — CHANGES_REQUESTED** [review]

File: src/graphql-server/resolvers.js (hunk 5 above)
Comment (JettTech) [reply src/graphql-server/resolvers.js]: Good catch. :+1:

**Review by robbiecarlton — APPROVED** [review]

**Pull request merged**